### DISCUSSION TITLE
API Routes: sizeLimit of bodyParser doesn't apply correctly in Vercel but does locally

### DISCUSSION THREAD

**Discussion by fonnebien** [vercel/next.js#56796] [Help] [answered]
### Summary

I'm experiencing a weird bug where I use an API route to handle uploading files.

Since the files can be larger than the default bodyParser size of 1mb, I increased it to 20mb to be sure the end user does not run into issues when uploading their file. (The file content is similar to a resume) 

When I apply the necessary change, like shown in the documentation, it works locally. Now when I deploy my application in Vercel, the upload feature does not work with a response status of 413 Payload too large.

Anyone else experiencing this?

### Additional information

```js
NextJS version: 12.0.7
React version: 17.0.2

The following piece of code is exported in the api route file.


export const config = {
  api: {
    bodyParser: {
      sizeLimit: '20mb',
    },
  },
};
```

**Comment by filipstev** [comment]
Any API route deployed on Vercel has a limit of 4.5mb. Locally, you're dealing with your own server, so you may upload bigger files. See [this link](https://vercel.com/guides/how-to-bypass-vercel-body-size-limit-serverless-functions)

**Reply by fonnebien** [reply, marked as answer]
Thank you for answering!

I already came to a solution in the meantime but lost track of this question so I didn't mark it as answered/solved. I stumbled upon the same link you attached and was able to bypass the 4.5MB limit by using Vercel Blob and client side uploads.
